File: server/plugin/inspection/router/production_order.go
package router

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

var ProductionOrder = new(productionOrderRouter)

type productionOrderRouter struct{}

func (r *productionOrderRouter) Init(private *gin.RouterGroup) {
	{
		group := private.Group("productionOrder").Use(middleware.OperationRecord())
		group.POST("submitDeviceData", apiProductionOrder.SubmitDeviceData)
		group.POST("createProductionOrder", apiProductionOrder.CreateProductionOrder)
		group.POST("confirmReworkReceived", apiProductionOrder.ConfirmReworkReceived)
		group.POST("confirmReworkDone", apiProductionOrder.ConfirmReworkDone)
		group.DELETE("deleteProductionOrder", apiProductionOrder.DeleteProductionOrder)
		group.DELETE("forceDeleteProductionOrder", apiProductionOrder.ForceDeleteProductionOrder)
		group.DELETE("deleteSubmittedDevice", apiProductionOrder.DeleteSubmittedDevice)
		group.PUT("updateProductionOrder", apiProductionOrder.UpdateProductionOrder)
		group.POST("assignBatch", apiProductionOrder.AssignBatch)
		group.POST("scanAssignBatch", apiProductionOrder.ScanAssignBatch)
		group.POST("createBatch", apiProductionOrder.CreateBatch)
	}
	{
		group := private.Group("productionOrder")
		group.GET("findProductionOrder", apiProductionOrder.FindProductionOrder)
		group.GET("getProductionOrderList", apiProductionOrder.GetProductionOrderList)
		group.GET("getSubmittedDeviceList", apiProductionOrder.GetSubmittedDeviceList)
		group.GET("findSubmittedDevice", apiProductionOrder.FindSubmittedDevice)
		group.GET("getDeviceStatusLogs", apiProductionOrder.GetDeviceStatusLogs)
	}
}

File: web/src/plugin/inspection/api/production_order.js
import service from '@/utils/request'

export const createProductionOrder = (data) => {
  return service({ url: '/productionOrder/createProductionOrder', method: 'post', data })
}

export const deleteProductionOrder = (params) => {
  return service({ url: '/productionOrder/deleteProductionOrder', method: 'delete', params })
}

export const forceDeleteProductionOrder = (params) => {
  return service({ url: '/productionOrder/forceDeleteProductionOrder', method: 'delete', params })
}

export const updateProductionOrder = (data) => {
  return service({ url: '/productionOrder/updateProductionOrder', method: 'put', data })
}

export const findProductionOrder = (params) => {
  return service({ url: '/productionOrder/findProductionOrder', method: 'get', params })
}

export const getProductionOrderList = (params) => {
  return service({ url: '/productionOrder/getProductionOrderList', method: 'get', params })
}

export const getSubmittedDeviceList = (params) => {
  return service({ url: '/productionOrder/getSubmittedDeviceList', method: 'get', params })
}

export const findSubmittedDevice = (params) => {
  return service({ url: '/productionOrder/findSubmittedDevice', method: 'get', params })
}

export const deleteSubmittedDevice = (params) => {
  return service({ url: '/productionOrder/deleteSubmittedDevice', method: 'delete', params })
}

export const confirmReworkDone = (data) => {
  return service({ url: '/productionOrder/confirmReworkDone', method: 'post', data })
}

export const confirmReworkReceived = (data) => {
  return service({ url: '/productionOrder/confirmReworkReceived', method: 'post', data })
}

export const scanAssignBatch = (data) => {
  return service({ url: '/productionOrder/scanAssignBatch', method: 'post', data })
}

export const getDeviceStatusLogs = (params) => {
  return service({ url: '/productionOrder/getDeviceStatusLogs', method: 'get', params })
}
